### DISCUSSION TITLE
Add ability for plugins to read/write files from disk

### DISCUSSION THREAD

**Discussion by Flyxxpy** [OpenRCT2/OpenRCT2#17089] [Feature requests]
Using external data in a plugin or saving plugin data is quite difficult.
The ability to read files from disk and write to the disk would make this process a lot easier, as it would, for example, allow to use data from PNG, CSV or JSON files.

**Comment by frutiemax** [comment]
As much as I am all in for scripting flexibility/modularity, this poses serious security threats. For CSV/Json, the data should be in the script file in a table.

For an image, one possibility would be to make sprite object types so that plugins can load the sprite object file and use the images contained in it. I'm against direct file write/read operations however.

**Reply by IntelOrca** [reply]
@frutiemax plugins will only be able to read/write files that the user has specifically chosen using a file dialog.
